### PR TITLE
Support for wolfSSL with OpenVPN v2.4.

### PR DESCRIPTION
**Pull request by dgarske**
wolfSSL:

Support added in: https://github.com/wolfSSL/wolfssl/pull/2503

```sh
git clone https://github.com/wolfSSL/wolfssl.git
cd wolfssl
./autogen.sh
./configure --enable-opensslall --enable-keygen --enable-rsapss --enable-des3 \
--enable-dtls --enable-certgen --enable-certreq --enable-certext \
--enable-sessioncerts --enable-crl --enable-ocsp \
CFLAGS="-DWOLFSSL_DES_ECB -DWOLFSSL_LOG_PRINTF -DWOLFSSL_PUBLIC_MP -DHAVE_EX_DATA"
make
sudo make install
```

OpenVPN:

```sh
autoreconf -i -v -f
./configure --with-crypto-library=wolfssl
make
make check
sudo make install
```

FYI: Also submitted patch via `git send-email --to=openvpn-devel@lists.sourceforge.net HEAD~1`

### CODE REVIEW

File: src/openvpn/options.c [hunk 1]
```diff
@@ -851,7 +851,11 @@ init_options(struct options *o, const bool init_gc)
     o->scheduled_exit_interval = 5;
 #endif
 #ifdef ENABLE_CRYPTO
+#ifdef ENABLE_CRYPTO_WOLFSSL
+    o->ciphername = "AES-256-CBC";
+#else
     o->ciphername = "BF-CBC";
+#endif
```
Comment (schwabe) [inline src/openvpn/options.c]: This is still a no go like I explained in my email to the first patch version

**Comment by dgarske** [comment]
Hi @schwabe , thanks for the note! The wolfSSL engineer @julek-wolfssl is going to work on this plus the other feedback previously sent. Thank you for taking the time to review.

File: src/openvpn/options.c (hunk 1 above)
Comment (schwabe) [reply src/openvpn/options.c]: to be clear, erroring out if no cipher is set would be okay but changing behaviour is not.

Comment (julek-wolfssl) [reply src/openvpn/options.c]: Hi, I proposed a solution to this in #134.

**Comment by julek-wolfssl** [comment]
The minimum configure options for wolfSSL (to make all OpenVPN tests pass) are:
`./configure --enable-opensslall --enable-des3 --enable-crl --enable-certgen --enable-certext --enable-aesctr --enable-sessioncerts CFLAGS="-DWOLFSSL_DES_ECB -DHAVE_EX_DATA"`

**Comment by dgarske** [comment]
Closing PR, since support has been upstreamed with https://github.com/OpenVPN/openvpn/pull/134

**Pull request closed without merging**